src/app/shared/components/graph-plotter/graph-plotter.component.ts

```diff
@@ -31,10 +32,22 @@ export class GraphPlotterComponent implements OnInit, AfterViewInit {
 		return this._properties;
 	}
 	@Input() set properties(properties: string[] | undefined) {
-		this._properties = properties;
-		this.plot();
+		if ((!this.properties && properties) || (this.properties && properties && this.properties.length !== properties.length)) {
```

Comment:
This predicate seems to be a bit long and is hard to understand. You could move the check to an operation with its own "speaking name" and documentation